webapp/src/app.tsx

```diff
@@ -5247,6 +5248,21 @@ export class ProjectView
 
     }
 
+    ///////////////////////////////////////////////////////////
+    ////////////             Navigate regions     /////////////
+    ///////////////////////////////////////////////////////////
+
+    hideNavigateRegions() {
+        this.setState({ navigateRegions: false });
+    }
+
+    showNavigateRegions() {
+        const dialog = Array.from(document.querySelectorAll("[role=dialog]")).find(dialog => (dialog as any).checkVisibility());
```

Comment:
checkVisibility isn't supported in all of the clients we target, but i wrote a polyfill for it a while back:

https://github.com/microsoft/pxt/pull/10241

probably need to move that polyfill up a bit so that it's applied before blockly is initialized, since this code might be called in a project with no blocks

Reply:
Good spot. I've generally been using offsetParent === null but that won't work here because the modals use `position: fixed`.

Reply:
Moved alongside app.tsx. I've kept it separate from the pxtsim polyfill code but could combine them. I didn't think the sim embed would need checkVisible.

99c5974e34abdbb253076ab5952c251a5c83b4b9

Incidentally, there might be quite a few polyfills for IE and very old Safari that can be removed from injectPolyphils (!) now.